src/soundsourceproxy.cpp

```diff
@@ -308,31 +255,163 @@ bool SoundSourceProxy::isFileSupported(const QFileInfo& fileInfo) {
 
 // static
 bool SoundSourceProxy::isFileNameSupported(const QString& fileName) {
-    return fileName.contains(getSupportedFileNameRegex());
+    return fileName.contains(getSupportedFileNamesRegex());
 }
 
 // static
 bool SoundSourceProxy::isFileExtensionSupported(const QString& fileExtension) {
-    return !s_soundSourceProviders.getProviderForFileExtension(fileExtension).isNull();
+    return !s_soundSourceProviders.getRegistrationsForFileExtension(fileExtension).isEmpty();
 }
 
 // static
-Mixxx::SoundSourcePointer SoundSourceProxy::initialize(
-        const QString& qFilename) {
-    const QUrl url(QUrl::fromLocalFile(qFilename));
-
-    const QString fileExtension(Mixxx::SoundSource::getFileExtensionFromUrl(url));
+Mixxx::SoundSourceProviderRegistrationList
+SoundSourceProxy::findSoundSourceProviderRegistrations(
+        const QUrl& url) {
+    QString fileExtension(Mixxx::SoundSource::getFileExtensionFromUrl(url));
     if (fileExtension.isEmpty()) {
-        qWarning() << "Unknown file type:" << qFilename;
-        return Mixxx::SoundSourcePointer();
+        qWarning() << "Unknown file type:" << url;
+        return Mixxx::SoundSourceProviderRegistrationList();
+    }
+
+    auto registrationsForFileExtension(
+            s_soundSourceProviders.getRegistrationsForFileExtension(
+                    fileExtension));
+    if (registrationsForFileExtension.isEmpty()) {
+        qWarning() << "Unsupported file type:" << url;
     }
 
-    Mixxx::SoundSourceProviderPointer pSoundSourceProvider(
-            s_soundSourceProviders.getProviderForFileExtension(fileExtension));
-    if (pSoundSourceProvider) {
-        return pSoundSourceProvider->newSoundSource(url);
+    return registrationsForFileExtension;
+}
+
+//Constructor
+SoundSourceProxy::SoundSourceProxy(
+        const QString& filePath,
+        SecurityTokenPointer pSecurityToken)
+        : m_filePath(filePath),
+          m_url(QUrl::fromLocalFile(m_filePath)),
+          m_pSecurityToken(openSecurityToken(m_filePath, pSecurityToken)),
+          m_soundSourceProviderRegistrations(findSoundSourceProviderRegistrations(m_url)),
+          m_soundSourceProviderRegistrationIndex(0) {
+    initSoundSource();
+}
+
+//Other constructor
+SoundSourceProxy::SoundSourceProxy(
+        const TrackPointer& pTrack)
+        : m_filePath(pTrack->getCanonicalLocation()),
+          m_url(QUrl::fromLocalFile(m_filePath)),
+          m_pTrack(pTrack),
+          m_pSecurityToken(openSecurityToken(pTrack)),
+          m_soundSourceProviderRegistrations(findSoundSourceProviderRegistrations(m_url)),
+          m_soundSourceProviderRegistrationIndex(0) {
+    initSoundSource();
```

Comment:
This way we have a single function that is responsible to init a working SoundSource